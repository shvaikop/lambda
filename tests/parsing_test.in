zero    = \f.   \x.  x
one = (\f.    \x.f x)
succ = \n.\f.\x. f (n f x)
two = $succ $one
three = $succ $two
plus = \m.\n.\f.\x.m f (n f x)
five = $plus $two $three
pred = \n.\f.\x.n (\g.\h.h (g f)) (\u.x) (\u.u)
four = $pred $five
$zero
$one
$two
$three
$four
$five
\m .\n.\f.\x.m f (n f x)