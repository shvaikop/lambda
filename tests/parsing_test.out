Up to date
Welcome to Lambda Calculus Interpreter
\f.\x.x
\f.\x.f x
\f.\x.(f) (f x)
\f.\x.(f) ((f) (f x))
\f.\x.(f) ((f) ((f) (f x)))
\f.\x.(f) ((f) ((f) ((f) (f x))))
"error" (line 1, column 3):
unexpected " "
expecting "."
